### PR TITLE
Interpolate healpix maps

### PR DESCRIPTION
**Pull request by keskitalo**
Add an operator that uses bilinear interpolation to generate continuous scanned signal from a HEALPix map. The maps are stored in node shared memory to reduce the required memory.

When scanning and co-adding multiple maps, they may be of different resolutions and some or all interpolated maps may be unpolarized.

### CODE REVIEW

**Review by tskisner — CHANGES_REQUESTED** [review]
Looks good, just a couple of small changes.

File: src/toast/ops/interpolate_healpix.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+    )
+
+    save_map = Bool(False, help="If True, do not delete map during finalize")
+
+    # FIXME : do we need this?
+    save_pointing = Bool(
+        False,
+        help="If True, do not clear detector pointing matrices if we "
+        "generate the pixel distribution",
+    )
```
[the remark this answers is not in the document]
Comment (keskitalo) [reply src/toast/ops/interpolate_healpix.py:87]: Thanks!

File: src/toast/ops/interpolate_healpix.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+        if self.stokes_weights is None or self.stokes_weights.mode == "I":
+            nnz = 1
+        elif self.stokes_weights.mode == "IQU":
+            nnz = 3
+        else:
+            msg = f"Unknown Stokes weights mode '{self.stokes_weights.mode}'"
+            raise RuntimeError(msg)
+
+        filenames = self.file.split(";")
+        detdata_keys = self.det_data.split(";")
```
[the remark this answers is not in the document]
Comment (keskitalo) [reply src/toast/ops/interpolate_healpix.py:169]: Done.

File: src/toast/ops/interpolate_healpix.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+
+            ob_data = data.select(obs_name=ob.name)
+            for idet, det in enumerate(dets):
+                self.detector_pointing.apply(ob_data, detectors=[det])
+                self.stokes_weights.apply(ob_data, detectors=[det])
+                det_quat = ob_data.obs[0].detdata[self.detector_pointing.quats][det]
+                # Convert pointing quaternion into angles
+                theta, phi, _ = qa.to_iso_angles(det_quat)
+                # Get pointing weights
+                weights = ob.detdata[self.stokes_weights.weights][det]
```
[the remark this answers is not in the document]
Comment (keskitalo) [reply src/toast/ops/interpolate_healpix.py:218]: That was sloppy, I agree. Fixed. `ref` should still reference the parent observation.

**Review by tskisner — APPROVED** [review]

File: src/toast/ops/interpolate_healpix.py [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+    subtract = Bool(
+        False, help="If True, subtract the map timestream instead of accumulating"
+    )
+
+    zero = Bool(False, help="If True, zero the data before accumulating / subtracting")
+
+    pixel_dist = Unicode(
+        "pixel_dist",
+        help="The Data key where the PixelDistribution object is located",
+    )
```
[the remark this answers is not in the document]
Comment (keskitalo) [reply src/toast/ops/interpolate_healpix.py]: Thanks! Removed.

**Pull request merged**